site/app/models/Team.php

```diff
@@ -202,6 +202,9 @@ public function getAssignmentSettings(Gradeable $gradeable) {
             $settings_file = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), "submissions", $gradeable->getId(), $this->getId(), "user_assignment_settings.json");
             $this->assignment_settings = FileUtils::readJsonFile($settings_file);
         }
-        return $this->assignment_settings;
+        if (isset($this->assignment_settings) && is_array($this->assignment_settings)) {
+            return $this->assignment_settings;
+        }
+        return ["team_history" => null];
```

Comment:
```suggestion
        return $this->assignment_settings ? $this->assignment_settings : ["team_history" => null];
```

This would protect against both `false` and an empty array (`[]`).

Reply:
That resolves to the same thing as $this->assignment_settings ? : ["team_history" => null]; right?

Reply:
Yeah, you're right. I always forget about the shorthand ternary statement within PHP.